src/main/java/net/apnic/rdap/rdap/RDAPObjectFactory.java

```diff
@@ -0,0 +1,54 @@
+package net.apnic.rdap.rdap;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Factory helper class for constructing conformant RDAP objects for client
+ * requests that are contextualised and pre filled with default notices.
+ */
+public class RDAPObjectFactory
+{
+    private static final Set<RDAPConformance> DEFAULT_CONFORMANCE = EnumSet.of(RDAPConformance.LEVEL_0);
+
+    private final List<RDAPNotice> defaultNotices;
+
+    public RDAPObjectFactory(List<RDAPNotice> defaultNotices) {
+        this.defaultNotices = Collections.unmodifiableList(
+                new ArrayList<>(Optional.ofNullable(defaultNotices).orElse(Collections.emptyList()))
+        );
+    }
+
+    /**
+     * Provides the default notices that have been set for every created
+     * RDAPObject subclass.
+     *
+     * @return List of default notices if set or null
+     */
+    public List<RDAPNotice> getDefaultNotices()
+    {
+        return defaultNotices;
+    }
+
+
+    /**
+     * Deep copies the default notices
+     */
+    private List<RDAPNotice> copyDefaultNoticesWithContext(String context)
```

Comment:
(Suggestion) Can this be taken out now that this should be immutable / unmodifiable?